Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -347,5 +378,79 @@ private bool TryInitializeCorrelationHelperIfNotInitialized()
                 return false;
             }
         }
+
+        /// <summary>
+        /// <see cref="System.Web.Handlers.TransferRequestHandler"/> can create a Child request to route extension-less requests to a controller.
+        /// (ex: site/home -> site/HomeController.cs)
+        /// We do not want duplicate telemetry logged for both the Parent and Child requests, so the activeRequests will be created OnBeginRequest.
+        /// When the child request OnEndRequest, the id will be removed from this dictionary and telemetry will not be logged for the parent.
+        /// </summary>
+        /// <remarks>
+        /// Unit tests should disable the ChildRequestTrackingSuppressionModule.
+        /// Unit test projects cannot create an [internal] IIS7WorkerRequest object.
+        /// Without this object, we cannot modify the Request.Headers without throwing a PlatformNotSupportedException.
+        /// Unit tests will have to initialize the RequestIdHeader.
+        /// The second IF will ensure the id is added to the activeRequests.
+        /// </remarks>
+        /// <remarks>
+        /// IIS Classic Pipeline should disable the ChildRequestTrackingSuppressionModule.
+        /// Classic does not create IIS7WorkerRequest object and Headers will be read-only.
+        /// (Exception System.PlatformNotSupportedException: This operation requires IIS integrated pipeline mode.)
+        /// </remarks>
+        private class ChildRequestTrackingSuppressionModule
+        {
+            /// <summary>
+            /// Max number of request ids to cache.
+            /// </summary>
+            private const int MAXSIZE = 100;
```

Comment:
Couple comments:
- 100 concurrent requests seems like an easy target to reach in heavily loaded applications, so the limit should be bumped;
- When the limit is reached (e.g. some `OnBegin` didn't have `OnEnd`, or some exception happened), it looks like we're clearing up all the active requests and `ShouldTrackRequest()` will decide not to track both requests (parent and child). Let's sync up offline on this and see if we can do something about it.